Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/workspace/ExternalWsDAO.java

```diff
@@ -0,0 +1,47 @@
+package edu.asu.spring.quadriga.dao.impl.workspace;
+
+import java.util.List;
+
+import org.hibernate.Query;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import edu.asu.spring.quadriga.dao.impl.BaseDAO;
+import edu.asu.spring.quadriga.dao.workspace.IExternalWsDAO;
+import edu.asu.spring.quadriga.dto.ExternalWorkspaceDTO;
+
+/**
+ * This class is responsible for Querying the MySQL database and fetch the class
+ * objects related to External Workspace.
+ *
+ */
+@Repository
+public class ExternalWsDAO extends BaseDAO<ExternalWorkspaceDTO> implements IExternalWsDAO {
+
+    @Autowired
+    private SessionFactory sessionFactory;
+
+    /**
+     * {@inheritDoc}
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    public ExternalWorkspaceDTO getExternalWorkspace(String externalWorkspaceid) {
+        ExternalWorkspaceDTO externalWorkspace = null;
+        Query query = sessionFactory.getCurrentSession()
+                .getNamedQuery("ExternalWorkspaceDTO.findByExternalWorkspaceid");
+        query.setParameter("externalWorkspaceid", externalWorkspaceid);
+        List<ExternalWorkspaceDTO> externalWorkspaceList = query.list();
```

Comment:
You should be able to use 'uniqueResult' here instead. Then you don't have to do the whole 'if list size > 0' check.